crates/daphne-server/src/storage_proxy_connection/kv/mod.rs

```diff
@@ -294,13 +332,18 @@ impl<'h> Kv<'h> {
         let key = Self::to_key::<P>(key);
 
         tracing::debug!(key, "PUT if not exists");
-        let response = self
+
+        let mut request = self
             .http
             .put(self.config.url.join(&key).unwrap())
             .bearer_auth(&self.config.auth_token)
-            .body(serde_json::to_vec(&value).unwrap())
-            .send()
-            .await?;
+            .body(serde_json::to_vec(&value).unwrap());
+
+        if let Some(expiration) = expiration {
+            request = request.header("X-Expiration", expiration.to_string());
```

Comment:
Let's make the header name a constant that we reuse rather than re-typing the raw string everywhere.

```rust
pub const STORAGE_PROXY_PUT_KV: &str = "...";
```

Reply:
there's a module for this as well `crates/daphne-service-utils/src/http_headers.rs` so you can share the constant between services